Create ui-c8y-1023-14-21-basic-settings-tab-should-be-hidden-if-basic-or-oauth-2.md (#4057)
SoftwareAG/c8y-docs: content/change-logs/platform-services/ui-c8y-1023-14-21-basic-settings-tab-should-be-hidden-if-basic-or-oauth-2.md

---
date: ""

> title: "Hide authentication's \"Basic settings\" tab if Basic or OAI-Secure login option is not configurable for the tenant"

Reviewer: ```suggestion
title: "Authentication tab hidden if corresponding login option is not configurable for the tenant"
```

> title: "Hide authentication's \"Basic settings\" tab if Basic or OAI-Secure login options are not available due to onlyManagementTenantAccess option"

Reviewer 2: ```suggestion
title: "Hide authentication's \"Basic settings\" tab if Basic or OAI-Secure login option is not configurable for the tenant"
```

product_area: Platform services
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-0UgqXH1Ys
    label: Administration
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-64347
version: 1023.14.21
---

> Since onlyManagementTenantAccess flag can be set for the Basic and OAI-Secure authentication methods, the Administration → Settings → Authentication → Basic settings tab will be shown or hidden depending on this flag:

Reviewer 3: Might i suggest re-wording as it is unlikely the average user will know what the onlyManagementTenantAccess flag is, nor can they do anything about it.  Does the following make sense:

In order to simplify the user experience changes have been made to the tabs available to users in the Administration --> Settings page.  If basic authentication is not enabled for the tenant the corresponding tab will not be shown.  This removes any confusion on what authentication methods are available for the tenant.

Further information on how authentication methods are configured can be found _please add the appropriate link_

> In order to simplify the user experience changes have been made to the tabs available to users in the Administration → Settings → Authentication page. If the tenant does not have permission to manage Basic or OAI-Secure login options, the corresponding tab will not be shown. This removes any confusion on what authentication methods are configurable for the tenant.

Reviewer: ```suggestion
To simplify the user experience, changes have been made to the tabs available to users in the Administration → Settings → Authentication page. If the tenant does not have permission to manage Basic or OAI-Secure login options, the corresponding tab will not be shown. This avoids confusion about what authentication methods are configurable for the tenant.
```

> Further information on how authentication methods are configured can be found in the [user documentation](https://cumulocity.com/docs/authentication/basic-settings).

Reviewer 2: ```suggestion
Further information on how to configure access settings for login options can be found in the [OpenAPI specification](https://cumulocity.com/api/core/#operation/putAccessLoginOptionResource).
```

> Further information on how to configure access settings for login options can be found in the [OpenAPI specification](https://cumulocity.com/api/core/#operation/putAccessLoginOptionResource).

Reviewer: ```suggestion
Further information on how to configure access settings for login options can be found in the [{{< openapi >}}](https://cumulocity.com/api/core/#operation/putAccessLoginOptionResource).
```

> - if onlyManagementTenantAccess is set to false for both (Basic or OAI-Secure) methods, the Basic settings tab will be visible to all relevant tenants.

Reviewer 2: ```suggestion
In order to simplify the user experience changes have been made to the tabs available to users in the Administration → Settings → Authentication page. If the tenant does not have permission to manage Basic or OAI-Secure login options, the corresponding tab will not be shown. This removes any confusion on what authentication methods are configurable for the tenant.

Further information on how authentication methods are configured can be found in the [user documentation](https://cumulocity.com/docs/authentication/basic-settings).
```

---
date: ""

> title: "Basic settings" tab should be hidden if `BASIC` or `OAUTH2_INTERNAL` login options throw 403 error (#10454) [GRAFT][release/y2026] (#10673)

Reviewer 2: ```suggestion
title: "Hide authentication's \"Basic settings\" tab if Basic or OAI-Secure login options are not available due to onlyManagementTenantAccess option"
```

product_area: Platform services
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-0UgqXH1Ys
    label: Administration
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-64347
version: 1023.14.21
---
More context info:
Depending on the value of the onlyManagementTenantAccess flag for the BASIC and OAI-Secure authentication methods, the Administration → Settings → Authentication → Basic settings tab should be dynamically shown or hidden, following the same behavior as already implemented for the Sign-on tab.

Criteria:
If onlyManagementTenantAccess is set to true for a given authentication method (BASIC or OAI-Secure), the Basic settings tab should be hidden for tenants other than the management tenant.
If onlyManagementTenantAccess is set to false for both (BASIC or OAI-Secure) methods, the Basic settings tab should be visible to all relevant tenants.

> onlyManagementTenantAccess directly affects if login option throws 403 error, therefore we check for this error when deciding whether to show the tab or not.

Reviewer 2: ```suggestion
Since onlyManagementTenantAccess flag can be set for the Basic and OAI-Secure authentication methods, the Administration → Settings → Authentication → Basic settings tab will be shown or hidden depending on this flag:
- if onlyManagementTenantAccess is set to true for an authentication method (Basic or OAI-Secure), the Basic settings tab will be hidden for tenants other than the management tenant
- if onlyManagementTenantAccess is set to false for both (Basic or OAI-Secure) methods, the Basic settings tab will be visible to all relevant tenants.
```
